### ISSUE TITLE
For rejected promises that were expected to be fulfilled, log the error that was thrown

### ISSUE THREAD

**Issue by JaKXz** [domenic/chai-as-promised#128]
```
  1) my project's component A should do something asynchronous:
     AssertionError: expected promise to be fulfilled but it was rejected with { Object (type, filename, ...) }

  2) my project's component B should do another asynchronous thing:
     AssertionError: expected promise to be fulfilled but it was rejected with { Object (type, filename, ...) }
```

It would be great if those printed the entire error object, just like when I add the catch block to my code to debug:

```coffeescript
.then (data) -> doStuff data
.catch (error) -> console.log error
```
Apologies if this has been discussed already / is implemented and I just can't find it on the docs.

**Comment by keithamus** [comment]
(@domenic this looks to be the same kind of issue as domenic/sinon-chai#67).

From a quick glance, I think we could get this diffing properly by just passing `true` as an additional argument into `assertion.assert` on lines [76 and 80](https://github.com/domenic/chai-as-promised/blob/master/lib/chai-as-promised.js#L75-L81). I'm not sure if any extra work is involved - but I'd say @JaKXz if you're up for making a PR, give that a go and see if it does what you want :smile:.

**Comment by JaKXz** [comment]
@keithamus I tried that locally - passing in true for the `Assertion.prototype.assert` method's `showDiff` parameter, but it didn't seem to work :( 

```js
        // These are for clarity and to bypass Chai refusing to allow `undefined` as actual when used with `assert`.
        function assertIfNegated(assertion, message, extra) {
            assertion.assert(true, null, message, extra.expected, extra.actual, true);
        }

        function assertIfNotNegated(assertion, message, extra) {
            assertion.assert(false, message, null, extra.expected, extra.actual, true);
        }
```

I also noticed that `config` API has `showDiff` set to true already, and I don't think I've changed it anywhere in my project, so unless my understanding of the "diff" message is wrong, I think there might be more work involved? Not 100% sure, will keep poking around.

**Comment by keithamus** [comment]
@JaKXz it's worth reading domenic/sinon-chai#67 as it offers a bit more context about the `showDiff` behaviour. If diffs aren't showing with `true` passed, then it could be that the reporter you're using doesn't output diffs. Which test runner and reporter are you using?

**Comment by tommysullivan** [comment]
I also have this problem using cucumber.js. 

During testing to see if it was cucumber.js or chai-as-promised, I had two test cases:

In one test case, I simply returned a rejected promise to cucumber.js with an Error instance. This was automatically printed to the output using the toString() method of the error instance, which I verified was called on my error instance by overriding its toString. Which is what I want, the full output of the error.

In the other test case, I used chai-as-promised like so: `expect(thePromise).to.eventually.be.fulfilled`. In this case what I get is the type of output shown in the original issue description.
